control-plane/api-gateway/controllers/gateway_controller.go

```diff
@@ -466,6 +472,10 @@ func SetupGatewayControllerWithManager(ctx context.Context, mgr ctrl.Manager, co
 			&source.Channel{Source: c.Subscribe(ctx, api.InlineCertificate, r.transformConsulInlineCertificate(ctx)).Events()},
 			&handler.EnqueueRequestForObject{},
 		).
+		Watches(
```

Comment:
we only care about when consul receives the JWTProvider so we only watch for the entry to come from consul